docs/device_automation_condition.md

```diff
@@ -18,10 +18,26 @@ The template will create a new file `device_condition.py` in your integration fo
 
 This is the schema for conditions. The base schema should be extended from `homeassistant.helpers.config_validation.DEVICE_CONDITION_BASE_SCHEMA`.
 
-#### `async def async_get_conditions(hass, device_id)`
+#### `async_get_conditions`
+
+```py
+async def async_get_conditions(
+    hass: HomeAssistant, device_id: str
+) -> list[dict[str, str]]:
+```
 
 Return a list of conditions that this device supports.
 
-#### `async def async_condition_from_config(config, config_validation)`
+#### `async_condition_from_config`
+
+```py
+@callback
+def async_condition_from_config(
+    config: ConfigType, config_validation: bool
+) -> condition.ConditionCheckerType:
+    """Create a function to test a device condition."""
+```
 
 Create a condition function from a function. The condition functions should be an async-friendly callback that evaluates the condition and returns a `bool`.
+
+Use the `config_valition` parameter to turn on or off config validation with the defined `CONDITION_SCHEMA`.
```

Comment:
This is not done by the integration developer, it's done by core depending on how the condiations are setup.

Reply:
Right.
```suggestion
The `config_valition` parameter will be used by the core to apply config validation conditionally with the defined `CONDITION_SCHEMA`.
```

Too much condition maybe? :smile:

Reply:
I think it's OK :)

Reply:
Should we apply my suggestion?

Reply:
Yes, sorry. `config_validation` is misspelled by the way.